Assets/Scripts/Models/Need.cs

```diff
@@ -83,25 +89,57 @@ public string DisplayAmount
         }
     }
 
+    /// <summary>
+    /// Gets the EventAction for the current furniture.
+    /// These actions are called when an event is called. They get passed the furniture
+    /// they belong to, plus a deltaTime (which defaults to 0).
+    /// </summary>
+    /// <value>The event actions that is called on update.</value>
+    public EventActions EventActions { get; private set; }
+
     // Update is called once per frame
     public void Update(float deltaTime)
     {
-        FunctionsManager.Need.CallWithInstance(luaUpdate, this, deltaTime);
-        if (luaOnly)
+        if (EventActions != null && EventActions.HasEvent("OnUpdate"))
         {
-            return;
+            EventActions.Trigger("OnUpdate", this, deltaTime);
         }
-
-        Amount += growthRate * deltaTime;
-
-        if (character != null && character.CurrTile.GetGasPressure("O2") < 0.15)
+        else 
         {
-            Amount += (addedInVacuum - (addedInVacuum * (character.CurrTile.GetGasPressure("O2") * 5))) * deltaTime;
+            DefaultNeedDecay(deltaTime);
         }
 
         if (Amount.AreEqual(100))
         {
-            // FIXME: Insert need fail damage code here.
+            if (EventActions != null && EventActions.HasEvent("OnEmptyNeed"))
+            {
+                EventActions.Trigger("OnEmptyNeed", this, deltaTime);
+            }
+            else
+            {
+                DefaultEmptyNeed();
+            }
+        } 
+        else if (Amount > 90f)
+        {
+            if (EventActions != null && EventActions.HasEvent("OnSevereNeed"))
```

Comment:
Sure. We don't have many needs yet to make a proper decision about Defaults. Maybe we won't even need them.

Since we will keep checking for the nullability of EventActions, I am ok with leaving the the other part.

I am not sure about a return value for trigger, since then it might not be clear if the return is because it run, or was the result of the action. It might need a new name if it does that.